torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -567,14 +541,8 @@ XLAGraphExecutor::SyncTensorCollection XLAGraphExecutor::CollectSyncTensors(
 }
 
 void XLAGraphExecutor::TensorCollectionBarrier(SyncTensorCollection* coll) {
-  static const std::string invalid_device(
-      "Unknown0"); /* Temp solution to identify unassigned devices */
-  if (coll->device.toString().compare(invalid_device) == 0 ||
-      coll->unlocker.size() > 0) {
-    return;
-  }
+  torch::lazy::LazyGraphExecutor::TensorCollectionBarrier(coll);
```

Comment:
hmm so the reason we need to overwrite this function is for future SPMD use case?

Reply:
Yea, or I can add the todo to the upstream. 😂